lib/ansible/modules/network/aci/msc_tenant.py

```diff
@@ -0,0 +1,182 @@
+  register: query_result
+'''
+
+RETURN = r'''
+'''
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils.network.aci.msc import MSCModule, msc_argument_spec, issubset
+
+
+def main():
+    argument_spec = msc_argument_spec()
+    argument_spec.update(
+        description=dict(type='str'),
+        display_name=dict(type='str'),
+        tenant=dict(type='str', required=False, aliases=['name', 'tenant_name']),
+        tenant_id=dict(type='str', required=False),
+        state=dict(type='str', default='present', choices=['absent', 'present', 'query']),
+    )
+
+    module = AnsibleModule(
+        argument_spec=argument_spec,
+        supports_check_mode=True,
+        required_if=[
+            ['state', 'absent', ['tenant']],
+            ['state', 'present', ['tenant']],
+        ],
+    )
+
+    description = module.params['description']
+    display_name = module.params['display_name']
+    tenant = module.params['tenant']
+    tenant_id = module.params['tenant_id']
+    state = module.params['state']
+
+    msc = MSCModule(module)
+
+    path = 'tenants'
+
+    # Query for existing object(s)
+    if tenant_id is None and tenant is None:
+        msc.existing = msc.query_objs(path)
+    elif tenant_id is None:
+        msc.existing = msc.get_obj(path, name=tenant)
+        if msc.existing:
+            tenant_id = msc.existing['id']
+    elif tenant is None:
+        msc.existing = msc.get_obj(path, id=tenant_id)
+    else:
+        msc.existing = msc.get_obj(path, id=tenant_id)
+        existing_by_name = msc.get_obj(path, name=tenant)
+        if existing_by_name and tenant_id != existing_by_name['id']:
+            msc.fail_json(msg="Provided tenant '{1}' with id '{2}' does not match existing id '{3}'.".format(tenant, tenant_id, existing_by_name['id']))
+
+    # If we found an existing object, continue with it
+    if tenant_id:
+        path = 'tenants/{id}'.format(id=tenant_id)
+
+    if state == 'query':
+        pass
+
+    elif state == 'absent':
+        msc.previous = msc.existing
+        if msc.existing:
+            if module.check_mode:
+                msc.existing = {}
+            else:
+                msc.existing = msc.request(path, method='DELETE')
+
+    elif state == 'present':
+        msc.previous = msc.existing
+
+        msc.sanitize(dict(
+            description=description,
+            id=tenant_id,
+            name=tenant,
+            displayName=display_name,
+            siteAssociations=[],
+            userAssociations=[dict(userId="0000ffff0000000000000020")],
+        ))
+
+        if msc.existing:
+            if not issubset(msc.sent, msc.existing):
+                if module.check_mode:
+                    msc.existing = msc.proposed
+                else:
+                    msc.existing = msc.request(path, method='PUT', data=msc.sent)
+        else:
+            if module.check_mode:
+                msc.existing = msc.proposed
+            else:
```

Comment:
msc.changed?

Reply:
Is taken care of by `fail_json()` and `exist_json()`.